.changeset/vast-paths-begin.md

```diff
@@ -0,0 +1,17 @@
+---
+"@fluidframework/datastore": minor
+"@fluidframework/test-runtime-utils": minor
+---
+---
+"section": legacy
+---
+
+Added ILayerCompatDetails property to FluidDataStoreRuntime, MockFluidDataStoreContext and MockFluidDataStoreRuntime
+
+An optional property called `ILayerCompatDetails` is added to `FluidDataStoreRuntime`. This is queried by
+`FluidDataStoreContext` in the `Runtime` layer to validate that the `Runtime` and `DataStore` layers are compatible.
+
+`ILayerCompatDetails` is also added to `MockFluidDataStoreRuntime` and `MockFluidDataStoreContext` which are used for
+testing because these mock classes are exported as legacy alpha.
+
+Note that this property is for internal use only.
```

Comment:
```suggestion
Important: this property is intended for use by Fluid Framework code only. No code outside the Fluid Framework should use or depend on this property in any way.
```

Am I right that this new property would ideally be `@internal`, but it can't be because it's a property on an otherwise public interface?

Reply:
yes, that's correct - It's a property on a legacy alpha class not interface.